community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/util/PathInterestFactory.java

```diff
@@ -0,0 +1,349 @@
+                {
+                    return STANDARD_COMPARATOR;
+                }
+            };
+
+    private static final PathInterest<? extends Comparable> ALL = new PathInterest<Comparable>()
+    {
+        @Override
+        public Comparator<Comparable> comparator()
+        {
+            return STANDARD_COMPARATOR;
+        }
+
+        @Override
+        public boolean canBeRuledOut( int numberOfVisits, Comparable pathPriority, Comparable oldPriority )
+        {
+            return false;
+        }
+
+        @Override
+        public boolean stillInteresting( int numberOfVisits )
+        {
+            return true;
+        }
+
+        @Override
+        public boolean stopAfterLowestCost()
+        {
+            return false;
+        }
+    };
+
+    public static <P extends Comparable> PathInterest<P> numberOfShortest( final int numberOfWantedPaths )
+    {
+        if ( numberOfWantedPaths < 1 )
+        {
+            throw new IllegalArgumentException( "Can not create PathInterest with interested in less than 1 path." );
+        }
+
+        return new VisitCountBasedPathInterest<P>()
+        {
+            private Comparator<P> comparator = new Comparator<P>()
+            {
+                @Override
+                public int compare( P o1, P o2 )
+                {
+                    return o1.compareTo( o2 );
+                }
+            };
+
+            @Override
+            int numberOfWantedPaths()
+            {
+                return numberOfWantedPaths;
+            }
+
+            @Override
+            public Comparator<P> comparator()
+            {
+                return comparator;
+            }
+        };
+    }
+
+    public static PathInterest<Double> allShortest( double epsilon )
+    {
+        return new PriorityBasedTolerancePathInterest( epsilon );
+    }
+
+    public static PathInterest<Double> all( double epsilon )
+    {
+        return new AllTolerancePathInterest( epsilon );
+    }
+
+    public static PathInterest<Double> numberOfShortest( double epsilon, int numberOfWantedPaths )
+    {
+        return new VisitCountBasedTolerancePathInterest( epsilon, numberOfWantedPaths );
+    }
+
+    public static PathInterest<Double> single( double epsilon )
+    {
+        return new SingleTolerancePathInterest( epsilon );
+    }
+
+    private static class PriorityBasedTolerancePathInterest extends PriorityBasedPathInterest<Double>
+    {
+        private final double epsilon;
+        private BiFunction<Double,Double,Boolean> interestFunction =
+                new BiFunction<Double,Double,Boolean>()
+                {
+                    @Override
+                    public Boolean apply( Double newValue, Double oldValue )
+                    {
+                        return NoneStrictMath.compare( newValue, oldValue, epsilon ) <= 0;
+                    }
+                };
+        private Comparator<Double> comparator = new Comparator<Double>()
```

Comment:
Same comparator used by lots of places. Does it make sense to define common in `NoneStrictMath`?